experimental/InjectiveResolutions/src/InjectiveResolutions.jl

```diff
@@ -0,0 +1,573 @@
+end
+struct IrrRes # irreducible resolution (including all computed data and the cochain complex)
+    irrSums::Vector{IrrSum}
+    cochainMaps::Vector{SubQuoHom}
+    surjections::Vector{SubQuoHom}
+    inclusions::Vector{SubQuoHom}
+    cokernels::Vector{SubquoModule}
+    cochainComplex::ComplexOfMorphisms # if sequence not exact return trivial cochain_complex (M0 -> M0)
+end
+
+##computes system of inequalities for polyhedron {r*a : r >= 0}, a \in ZZ^2
+##INPUT:    vector a in ZZ^2
+##OUTPUT:   matrix A, vector b such that Ax <= b for all x on hyperplane through a
+function compute_inequalities(a::Vector{Int})
+    num_variables = length(a)
+    num_inequalities = 2 * num_variables
+
+    A = zeros(Int, num_inequalities, num_variables)
+    b = zeros(Int, num_inequalities)
+    if !is_zero(a)
+        A[1,1] = -1
+        A[1,2] = 0
+        A[2,1] = 0
+        A[2,2] = -1
+        A[3,1] = a[2]
+        A[3,2] = -a[1]
+        A[4,1] = -a[2]
+        A[4,2] = a[1]
+    else
+        A[1,1] = -1
+        A[1,2] = 0
+        A[2,1] = 0
+        A[2,2] = -1
+        A[3,1] = 1
+        A[3,2] = 0
+        A[4,1] = 0
+        A[4,2] = 1
+    end
+    return A, b
+end
+
+@doc raw"
+    get_face_to_prime(P_F::Ideal)
+
+    Given a homogeneous prime ideal, return the corresponding face F
+
+    INPUT:    prime ideal p_F
+    OUTPUT:   list [face F, matrix A, vector b] such that F is defined by Ax <= b
+"
+function get_face_to_prime(P_F::Ideal)
+    R_Q = base_ring(P_F)
+    #get generators of k[F] = k[Q]/P_F
+    Q_F,_ = quo(R_Q,P_F)
+    G_F = gens(Q_F)
+    L_F = (P_F,[])
+    for g in G_F
+        if !is_zero(g)
+            push!(L_F[2], degree(Vector{Int},g))
+        end
+    end
+    
+    #get corresponding face defining it by a system of inequalities
+    F_poly = []
+    m = length(gens(grading_group(R_Q)))
+    C = convex_hull(zeros(Int, m))
+    A = Nothing
+    b = Nothing
+    for a in L_F[2]
+        A,b = compute_inequalities(a)
+        C_a = polyhedron(A,b)
+        C = C + C_a
+        push!(F_poly,[C,A,b])
+    end
+    if L_F[2] == Any[]
+        A = zeros(Int, m,m)
+        b = zeros(Int, m)
+        push!(F_poly,[C,A,b])
+    end
+    return F_poly
+end
+
+##get all prime ideals P_F associated to M
+##INPUT:    ideal I
+##OUTPUT:   list of all minimal associated primes + the maximal ideal and the corresponding face F as a polyhedron  
+function get_all_ass_primes(I)
+    R_Q = base_ring(I)
+    I_m = ideal(R_Q,gens(R_Q)) #maximal ideal
+    F_m = get_face_to_prime(I_m)[1]
+    P_F = [FaceQ(I_m,F_m[1],F_m[2],F_m[3])]
+    MP = minimal_primes(I)
+    for mp in MP
+        p_f = get_face_to_prime(mp)[1]
+        push!(P_F,FaceQ(mp,p_f[1],p_f[2],p_f[3]))
+    end
+    return P_F
+end
+
+
+##Let W = k{Q\(a + F - Q)} and let H be a hyperplane containing F as a subset.
+##Calculates generators of k{(a + H_+^°)\cap Q}.
+##INPUT:    semigroup Q_P as a polyhedron
+#           zonotope G_Q of Q_P as in Lemma 3.10
```

Comment:
Excellent: I really appreciate these references to specific lemmas and theorems in the code, that will surely be a great asset to people in the future!

One tiny request, though: it is not immediately clear to me to which book or article these reference -- presumably https://arxiv.org/abs/math/0309256 from the PR description? If so, it would be great if e.g. at the top of this file, or in the README (or both places) this link could appear to remove any ambiguity (also, once this PR is merged, someone might read the code w/o knowing which PR it came from, so they wouldn't about that arXiv link at all.

(long term we'd put this paper into our bib file, too, but no need to do that now unless you really want to for some reason.)